corral/util/log.pony

```diff
@@ -99,16 +19,27 @@ primitive CodeLogFormatter is LogFormatter
      .> append(": ")
      .> append(msg)
 
-primitive LevelLogFormatter is LogFormatter
-  fun apply(level: LogLevel, msg: String, loc: SourceLoc): String =>
-    (recover String(level.string().size()
-      + msg.size()
-      + 4)
-    end)
-     .> append(level.string())
-     .> append(": ")
-     .> append(msg)
+primitive LevelLogFormatter
+  fun apply(level: LogLevel): LogFormatter =>
+    let level_string = match level
+    | Fine => "FINE"
+    | Info => "INFO"
+    | Warn => "WARN"
+    | Error => "ERRR"
+    end
+    recover val
+      object
+        fun apply(msg: String, loc: SourceLoc): String =>
+          (recover String(level_string.size()
+            + msg.size()
+            + 4)
+          end)
+          .> append(level_string)
+          .> append(": ")
+          .> append(msg)
+      end
+    end
 
 primitive SimpleLogFormatter is LogFormatter
-  fun apply(level: LogLevel, msg: String, loc: SourceLoc): String =>
```

Comment:
I assume the log level where was so it would be polymorphic. Did you see that there's no need for being able to switch between these formatters?